packages/headless/src/features/commerce/common/actions.ts

```diff
@@ -65,7 +65,9 @@ export const buildBaseCommerceAPIRequest = (
     organizationId: state.configuration.organizationId,
     trackingId: state.configuration.analytics.trackingId,
     ...restOfContext,
-    clientId: navigatorContext.clientId,
+    clientId: state.configuration.analytics.enabled
```

Comment:
@Spuffynism does the commerce API now allow for the empty string to be passed as a clientId? I know this was making requests fail previously...

Reply:
Yeah this is fine since https://github.com/coveo-platform/commerce-service/pull/2069

Reply:
Would it be cleaner/more consistent to use the pattern on line 72, not sending a key with an empty string? Same comment for the query suggest,

```
...(state.configuration.analytics.enabled ? {clientId: navigatorContext.clientId} : {})
```

Reply:
You're right, that would be cleaner @samisayegh  👍 
FYI @fbeaudoincoveo , @Spuffynism , that does mean that I'll make the `clientId` optional (see `commerce-api-params.ts`), let me know if that's an issue.
Update incoming :loading: